File: CF Python/L.py
k = int(input())
n = int(input())
count = {x: {} for x in range(1, k + 1)}
count_x = {x: 0 for x in range(1, k + 1)}
x_y = {x: set() for x in range(1, k + 1)}

for i in range(n):
    x, y = map(int, input().split())
    count_x[x] += 1
    if y in x_y[x]:
        count[x][y] += 1
    else:
        count[x][y] = 1
        x_y[x].add(y)

dy = 0
for x in range(1, k + 1):
    ey = 0
    ey2 = 0
    for y in x_y[x]:
        r = (count[x][y] / count_x[x])
        ey += r * y
        ey2 += r * y * y
    dy += (ey2 - ey ** 2) * (count_x[x] / n)

print(dy)

File: CF PABC.NET/L.pas
begin
  var (K, N) := ReadlnInteger2();
  var Count := Range(1, K).ToDictionary(K -> K, V -> new Dictionary<Integer, Integer>());
  var CountX := Range(1, K).ToDictionary(K -> K, V -> 0);
  var XY := Range(1, K).ToDictionary(K -> K, V -> new HashSet<Integer>());
  
  for var I := 1 to N do
  begin
    var (X, Y) := ReadlnInteger2();
    CountX[X] += 1;
    if Y in XY[X] then
      Count[X][Y] += 1
    else begin
      Count[X][Y] := 1;
      XY[X] += [Y];
    end;
  end;
  
  Range(1, K).Sum(X -> begin
    var eY, eY2: Double;
    XY[X].ForEach(Y -> begin
      var R := Count[X][Y] * 1.0 / CountX[X];
      eY += R * Y;
      eY2 += R * Y * Y;
    end);
    Result := (eY2 - eY ** 2) * (CountX[X] * 1.0 / N);
  end).Println();
end.